File: back-end/src/main/java/com/example/Model/Destination.java
package com.example.Model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "Destination")
public class Destination {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long destinationID;

    @Column(name = "geolocation")
    private String geolocation;

    @Column(name = "title")
    private String title;

    @Column(name = "image")
    private String image;

    @Column(name = "description")
    private String description;

    @Column(name = "fromDate")
    private LocalDate fromDate;

    @Column(name = "toDate")
    private LocalDate toDate;
}

File: front-end/bucketlist-fe/src/app/features/Models/destination.ts
import { DatePipe } from "@angular/common";

export interface DestinationDTO {
    destinationID: number;
    geolocation: string;
    title: string;
    image: string;
    description: string;
    fromDate:DatePipe;
    toDate:DatePipe;
}

export interface DestinationNoIDDTO {
    geolocation: string;
    title: string;
    image: string;
    description: string;
}